style(app): App style design feedback slideout factory reset

koji opened the pull request:
<!--
Thanks for taking the time to open a pull request! Please make sure you've read the "Opening Pull Requests" section of our Contributing Guide:

https://github.com/Opentrons/opentrons/blob/edge/CONTRIBUTING.md#opening-pull-requests

To ensure your code is reviewed quickly and thoroughly, please fill out the sections below to the best of your ability!
-->

# Overview
This PR will fix Factory Slideout design feedback and update checkbox field styling.

**In terms of banner styling, I need Mel's help to understand what she pointed out exactly.**

<!--
Use this section to describe your pull-request at a high level. If the PR addresses any open issues, please tag the issues here.
-->

# Changelog
- Update Factory Reset slideout styling
- Add text style to checkboxField

<!--
List out the changes to the code in this PR. Please try your best to categorize your changes and describe what has changed and why.

Example changelog:
- Fixed app crash when trying to calibrate an illegal pipette
- Added state to API to track pipette usage
- Updated API docs to mention only two pipettes are supported

IMPORTANT: MAKE SURE ANY BREAKING CHANGES ARE PROPERLY COMMUNICATED
-->

# Review requests

<!--
Describe any requests for your reviewers here.
-->

# Risk assessment
low
<!--
Carefully go over your pull request and look at the other parts of the codebase it may affect. Look for the possibility, even if you think it's small, that your change may affect some other part of the system - for instance, changing return tip behavior in protocol may also change the behavior of labware calibration.

Identify the other parts of the system your codebase may affect, so that in addition to your own review and testing, other people who may not have the system internalized as much as you can focus their attention and testing there.
-->

koji committed:
style(app): this PR will fix Factory Slideout design feedback

This PR will fix Factory Slideout design feedback and update checkboxField styling

fix #10618

jerader commented on app/src/organisms/Devices/RobotSettings/AdvancedTab/AdvancedTabSlideouts/FactoryResetSlideout.tsx:
```diff
@@ -10,7 +10,7 @@ import {
   TYPOGRAPHY,
   Link,
-  CheckboxField,
+  // CheckboxField,
```
this should be deleted

jerader commented on app/src/atoms/CheckboxField/index.tsx:
```diff
@@ -90,6 +90,19 @@ const INNER_STYLE_NO_VALUE = css`
+  font-weight: ${TYPOGRAPHY.fontWeightRegular};
+  color: ${COLORS.darkBlack};
+  flex: 0 0 auto;
+  padding: 0 0.5rem;
```
this can be `0 ${SPACING.spacing3}`;

jerader reviewed (approved):
just a couple of comments but lgtm!

mmencarelli commented:
Round 2 feedback on factory reset slideout 

- [ ] (1) Up the size of this icon to 24px, currently it's 20px 
- [ ] (2) Wondering why this is a different style from design? 
- [ ] (3) Add 8px margin between the list items
- [ ] (4) Looks like there's a barely noticeable grey line right underneath the button? 
- [ ] (5) remove box shadow from `.layXKe`
- [ ] (6) Looks like the SVG is a bit thicker than the designs—when you get to this step, contact me and I can try sending you another SVG
- [ ] (7) The hover state colour should be our darkGreyHover (#646468), currently it's #283d52
- [ ] (8) When I try to tab through all the links using my keyboard none of the links of the tabs go into their focusVisible state


<img width="611" alt="Screen Shot 2022-06-21 at 3 25 24 PM" src="https://user-images.githubusercontent.com/54682054/174907002-e2077252-120c-41b4-82e9-6e1a72b45112.png">

koji committed:
sync up with release_6.0.0
Merge branch 'release_6.0.0' into app_style-design-feedback-slideout-factory-reset

koji committed:
Update index.tsx
